src/storage/mvcc/txn.rs

```diff
@@ -48,12 +47,24 @@ impl<S: Snapshot> MvccTxn<S> {
             // IsolationLevel is `Si`, actually the method we use in MvccTxn does not rely on
             // isolation level, so it can be any value.
             reader: MvccReader::new(snapshot.clone(), None, fill_cache, IsolationLevel::Si),
-            gc_reader: MvccReader::new(
-                snapshot,
-                Some(ScanMode::Forward),
-                fill_cache,
-                IsolationLevel::Si,
-            ),
+            start_ts,
+            writes: vec![],
+            write_size: 0,
+            collapse_rollback: true,
+        })
+    }
+    pub fn for_scan(
```

Comment:
How about combine for_scan and new?

Reply:
The MvccReader was used universally in testcase.  Change the interface would make difference to too much code.